src/system/SystemPacketBuffer.cpp

```diff
@@ -333,14 +333,18 @@ PacketBuffer* PacketBuffer::Consume(uint16_t aConsumeLength)
 {
     PacketBuffer* lPacket = this;
 
-    while (lPacket != NULL && aConsumeLength > 0)
+    while (aConsumeLength > 0)
```

Comment:
I'm not strongly against this change, but its not adding any significant value.  The change merely moves the exit test from one point in the loop to another, under the assumption that lPacket should never be NULL when the loop is entered.   As I see it, it doesn't make the code appreciably faster, smaller, easier to read or more correct.

Reply:
I understand better now how the compiler was moving the check to the bottom of the loop anyway, and that this change merely made that "optimization" explicit.  Again, I fine with this change.